test/strategy/StargateV2Strategy.t.sol

```diff
@@ -0,0 +1,725 @@
+        vm.expectRevert("Ownable: caller is not the owner");
+        strategy.removePToken(0);
+    }
+
+    function test_RevertWhen_CollateralAllocated() public useKnownActor(USDS_OWNER) {
+        AssetData memory data = assetData[0];
+
+        // Mock asset allocation!
+        stdstore.target(address(strategy)).sig("assetInfo(address)").with_key(data.asset).depth(0).checked_write(1e18);
+
+        (uint256 allocatedAmt,) = strategy.assetInfo(data.asset);
+
+        assert(allocatedAmt > 0);
+        vm.expectRevert(abi.encodeWithSelector(CollateralAllocated.selector, data.asset));
+        strategy.removePToken(0);
+    }
+}
+
+contract Test_ChangeSlippage is StargateStrategyV2Test {
+    uint16 public updatedDepositSlippage = 100;
+    uint16 public updatedWithdrawSlippage = 200;
+
+    function setUp() public override {
+        super.setUp();
+        vm.startPrank(USDS_OWNER);
+        _initializeStrategy();
+        vm.stopPrank();
+    }
+
+    function test_UpdateSlippage() public useKnownActor(USDS_OWNER) {
+        vm.expectEmit(address(strategy));
+        emit SlippageUpdated(updatedDepositSlippage, updatedWithdrawSlippage);
+        strategy.updateSlippage(updatedDepositSlippage, updatedWithdrawSlippage);
+        assertEq(strategy.depositSlippage(), updatedDepositSlippage);
+        assertEq(strategy.withdrawSlippage(), updatedWithdrawSlippage);
+    }
+
+    function test_RevertWhen_NotOwner() public {
+        vm.expectRevert("Ownable: caller is not the owner");
+        strategy.updateSlippage(updatedDepositSlippage, updatedWithdrawSlippage);
+    }
+
+    function test_RevertWhen_slippageExceedsMax() public useKnownActor(USDS_OWNER) {
+        vm.expectRevert(abi.encodeWithSelector(Helpers.GTMaxPercentage.selector, 10001));
+        strategy.updateSlippage(10001, 10001);
+    }
+}
+
+contract Test_Deposit is StargateStrategyV2Test {
+    function setUp() public override {
+        super.setUp();
+        vm.startPrank(USDS_OWNER);
+        _initializeStrategy();
+        _setAssetData();
+        vm.stopPrank();
+    }
+
+    function testFuzz_Deposit(uint256 amount) public useKnownActor(VAULT) {
+        amount = uint256(bound(amount, 1, 1e10));
+        for (uint8 i = 0; i < assetData.length; ++i) {
+            uint256 scaledAmt = amount * 10 ** ERC20(assetData[i].asset).decimals();
+            deal(assetData[i].asset, VAULT, scaledAmt, true);
+            ERC20(assetData[i].asset).approve(address(strategy), scaledAmt);
+
+            strategy.deposit(assetData[i].asset, scaledAmt);
+
+            assertEq(strategy.checkBalance(assetData[i].asset), scaledAmt);
+            uint256 _bal = ERC20(assetData[i].asset).balanceOf(address(strategy));
+            emit log_named_uint("Strategy Balance", _bal);
+            assertApproxEqAbs(ERC20(assetData[i].asset).balanceOf(address(strategy)), 0, 1);
+        }
+    }
+
+    function test_Deposit() public useKnownActor(VAULT) {
```

Comment:
why do we have this if we already have the fuzz version. i think this can be removed